src/main/java/org/springframework/cloud/stream/app/plugin/utils/MavenModelUtils.java

```diff
@@ -218,12 +220,18 @@ public static void addDockerPlugin(String artifactId, String version, String doc
         final Xpp3Dom entryPoint = new Xpp3Dom("entryPoint");
         build.addChild(entryPoint);
 
-        final Xpp3Dom exec = new Xpp3Dom("exec");
-        entryPoint.addChild(exec);
-
-        addElement(exec, "arg", "java");
-        addElement(exec, "arg", "-jar");
-        addElement(exec, "arg", "/maven/" + artifactId + ".jar");
+        if (ENTRYPOINT_TYPE_SHELL.equalsIgnoreCase(entrypointStyle)) {
+            final Xpp3Dom shell = new Xpp3Dom(ENTRYPOINT_TYPE_SHELL);
+            shell.setValue("$JAVA_HOME/bin/java $JAVA_OPTS -jar /maven/" + artifactId + ".jar ${*}");
```

Comment:
@gitlabbin What controls the substitution on `${*}` here? For example with straight docker one could enable boot debug by:

`docker run issuetestimage -c --debug`

Reply:
i understand what `${*}` does, i was asking for example usage given the entry style usage description: "For exec app properties should be passed in as command line args, for shell they should be passed in as environment variables"